scripts/welding_schema.py

```diff
@@ -103,119 +103,118 @@ def single_pass_weld_example(out_file="single_pass_weld_example.asdf"):
         name="Calibration current measurement",
-        input_signal=current_AD_transform.output_signal,
-        output_signal=msm.Signal("digital", "A", data=current_data),
         error=msm.Error(0.0),
-        func=current_calib_func,
+        func=MathematicalExpression(
+            "a * x + b", dict(a=Q_(1000.0 / 32768.0, "A"), b=Q_(0.0, "A"))
+        ),
     )
-    current_calib_transform.wx_metadata = dict(software=twincat_scope)
 
-    welding_current_chain = msm.MeasurementChain(
-        name="welding current measurement chain",
-        data_source=src_current,
-        data_processors=[current_AD_transform, current_calib_transform],
-    )
+    # define voltage source and transformations
 
-    welding_current = msm.Measurement(
-        name="welding current measurement",
-        data=[current_data],
-        measurement_chain=welding_current_chain,
-    )
-
-    src_voltage = msm.Source(
+    src_voltage = msm.SignalSource(
         name="Voltage Sensor",
         output_signal=msm.Signal("analog", "V", data=None),
         error=msm.Error(Q_(0.1, "percent")),
     )
 
-    HKS_sensor.sources.append(src_voltage)
-
-    # define AD conversion expression and transformation step
-    [a, x, b] = sympy.symbols("a x b")
-    voltage_ad_func = MathematicalExpression(a * x + b)
-    voltage_ad_func.set_parameter("a", Q_(32768.0 / 10.0, "1/V"))
-    voltage_ad_func.set_parameter("b", Q_(0.0, ""))
-
-    voltage_AD_transform = msm.DataTransformation(
+    voltage_AD_transform = msm.SignalTransformation(
         name="AD conversion voltage measurement",
-        input_signal=src_voltage.output_signal,
-        output_signal=msm.Signal("digital", "", data=None),
         error=msm.Error(Q_(0.01, "percent")),
-        func=voltage_ad_func,
+        func=MathematicalExpression(
+            "a * x + b", dict(a=Q_(32768.0 / 10.0, "1/V"), b=Q_(0.0, ""))
+        ),
+        type_transformation="AD",
     )
 
-    HKS_sensor.data_transformations.append(voltage_AD_transform)
-
-    # define voltage output calibration expression and transformation
-    voltage_calib_func = MathematicalExpression(a * x + b)
-    voltage_calib_func.set_parameter("a", Q_(100.0 / 32768.0, "V"))
-    voltage_calib_func.set_parameter("b", Q_(0.0, "V"))
-
-    voltage_calib_transform = msm.DataTransformation(
+    voltage_calib_transform = msm.SignalTransformation(
         name="Calibration voltage measurement",
-        input_signal=voltage_AD_transform.output_signal,
-        output_signal=msm.Signal("digital", "V", data=voltage_data),
         error=msm.Error(0.0),
-        func=voltage_calib_func,
+        func=MathematicalExpression(
+            "a * x + b", dict(a=Q_(100.0 / 32768.0, "V"), b=Q_(0.0, "V"))
+        ),
     )
+
+    # Define lab equipment
+
+    HKS_sensor = msm.GenericEquipment(
+        name="HKS P1000-S3",
+        sources=[src_current, src_voltage],
+        transformations=[voltage_AD_transform],  # Is this really done by this eq.?
```

Comment:
Is the voltage AD conversion done by the HKS sensor?

Reply:
no the HKS sensor outputs 0V - 10V (`"AA"`), `"AD"` happens at `ELM_3104`

Reply:
As I thought.